flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/source/IcebergSource.java

```diff
@@ -205,12 +225,35 @@ private SplitEnumerator<IcebergSourceSplit, IcebergEnumeratorState> createEnumer
         // Only do scan planning if nothing is restored from checkpoint state
         List<IcebergSourceSplit> splits = planSplitsForBatch(planningThreadName());
         assigner.onDiscoveredSplits(splits);
+        // clear the cached splits after enumerator creation as they won't be needed anymore
+        this.batchSplits = null;
       }
 
       return new StaticIcebergEnumerator(enumContext, assigner);
     }
   }
 
+  boolean shouldInferParallelism() {
+    return !scanContext.isStreaming();
+  }
+
+  int inferParallelism(ReadableConfig flinkConf, StreamExecutionEnvironment env) {
```

Comment:
nit: could this be private?

Reply:
good catch. it was package visible as earlier implementation does the inferring in the `IcebergTableSource`. with the refactoring and `buildStream` introduced. this can be private.